docs/prepare-image-animation.md

```diff
@@ -0,0 +1,165 @@
+
+#### `srcImg`
+
+An `<img>` to animate from. This is used to determine the position, size, and the `object-fit` property to start the animation with.
+
+
+#### `targetImg`
+
+An `<img>` to animate to. This is used to determine the position, size, and the `object-fit` property to end the animation with.
+
+#### `srcImgRect`
+
+Defaults to `srcImg.getBoundingClientRect()`. If the `srcImg` is not laid out at the time you call `prepareImageAnimation`, you will want to capture the `ClientRect` beforehand and provide it to the call.
+
+One situtation this might be useful is if you are doing an animation between pages, where the content is in the `body` itself rather than in a separate scrolling container. For example. consider the following page structure:
+
+```html
+<body>
+  <div class="page">
+    …
+  </div>
+  <div class="page" hidden>
+    <h1>Some title that might wrap depending on the viewport width</h1>
+    <img class="hero" …>
+  </div>
+</body>
+```
+
+To figure out where the `hero` will be positioned, we need to layout the target page (e.g. by adding `hidden` to the current page and removing it from the target page). However, hiding the current page will mean `prepareImageAnimation` will no longer know where to start the animation. By providing `srcImgRect`, the animation can know where to start from.
+
+#### `targetImgRect`
+
+Defaults to `targetImg.getBoundingClientRect()`. If you know where the `targetImg` will be rendered, but you have not laid out the containing content, you can provide it to `prepareImageAnimation`. You can use this to avoid a forced layout in some situations, for example in the [hero animation demo](./demo/hero), we do something like:
+
+```javascript
+// Layout the the target so that we know where targetImg is
+target.hidden = false;
+
+// Forced style calc + layout when we go to measure things
+const {
+  applyAnimation,
+  cleanupAnimation,
+} = prepareImageAnimation(…);
+
+// Regular style calc + layout for mutations
+current.hidden = true;
+applyAnimation();
+```
+
+The forced style calculation caused by `prepareImageAnimation` can be avoided if you already know where `targetImg` will be positioned. Note that in this case, you will still need to provide a `targetImg` to the function so that the animation knows the `object-fit` property to animate to.
+
+#### `curve`
+
+This option defaults to `{x1: 0.42, y1: 0, x2: 0.58, y2: 1}`, which is the same as the built-in `ease-in-out` transition timing function. This is an object with the control points for a [`cubic-bezier()`](https://developer.mozilla.org/en-US/docs/Web/CSS/single-transition-timing-function#The_cubic-bezier()_class_of_timing_functions) curve and is used to determine the animation progress for the position, size and crop at any given time.
```

Comment:
This option defaults to the built-in `ease-in-out` transition timing function (`{x1: 0.42, y1: 0, x2: 0.58, y2: 1}`).